languages/rust/exercises/concept/string-vs-str/.meta/design.md

```diff
@@ -0,0 +1,50 @@
+# Goal
+
+Teach the student the differences between String and &str and how to work successfully with both.
+Learning objectives
+
+    Know String is owned and mutable
+    Know &str is shared and immutable
+    Know String has &str methods because of shared traits
+    Know of performance characteristics
+
+## Out of scope
+
+    Machine level representation of either
```

Comment:
Disagree with this assertion: the property "All strings are valid utf8" is essential to understand why indexing isn't implemented, for example.